ui/src/main/java/com/midtrans/sdk/ui/models/PaymentResult.java

```diff
@@ -0,0 +1,216 @@
+package com.midtrans.sdk.ui.models;
+
+import android.text.TextUtils;
+
+import com.midtrans.sdk.core.models.snap.BaseTransactionResponse;
+import com.midtrans.sdk.core.models.snap.card.CreditCardPaymentResponse;
+import com.midtrans.sdk.ui.constants.Payment;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Created by ziahaqi on 2/20/17.
+ */
+
+public class PaymentResult implements Serializable {
```

Comment:
We can make this general by using generic.

```java
public class PaymentResult<T extends from BaseTransactionResponse>
```

Then we can specific transaction response in here.